mlflow/spark/__init__.py

```diff
@@ -1106,6 +1107,20 @@ def autolog(disable=False, silent=False):  # pylint: disable=unused-argument
     )
     from mlflow.utils._spark_utils import _get_active_spark_session
 
+    # Check if environment variable PYSPARK_PIN_THREAD is set to false.
+    # The "Pin thread" concept was introduced since Pyspark 3.0.0 and set to default to true
+    # since Pyspark 3.2.0 (https://issues.apache.org/jira/browse/SPARK-35303). When pin thread
+    # is enabled, Pyspark manages Python and JVM threads in a 1:1, meaning that when one thread
+    # is terminated, the corresponding thread in the other side will be terminated as well.
+    # However, this causes an issue in spark autologging as our event listener thread may be
+    # terminated before receiving the datasource event.
+    # Hence, we have to disable it and decouple the thread management between Python and JVM.
+    if pyspark_version >= "3.2" and os.environ.get("PYSPARK_PIN_THREAD", "").lower() != "false":
```

Comment:
Comparing versions as a string can lead to errors. Can we use `packaging`?

Reply:
Sure! Fixing now.